califa/CMakeLists.txt

```diff
@@ -83,3 +86,5 @@ endif(FAIRDB_FOUND)
 
 
 GENERATE_LIBRARY()
+target_link_libraries(R3BCalifa boost_regex)
```

Comment:
boost_regex should go into set(DEPENDENCIES

Reply:
`[ 74%] Linking CXX shared library ../lib/libtimestitcher.so
825/usr/bin/ld: cannot find -lboost_regex
826timestitcher/CMakeFiles/timestitcher.dir/build.make:150: recipe for target 'lib/libtimestitcher.so.SOVERSION' failed
`
Note that this is not for califa, but for libtimestitcher.so. 
It seems dependencies get added transitively, while link directories don't.

Reply:
the issue seems to be that target_link_libraries, as called by  $FAIRROOTPATH/share/fairbase/cmake/modules/ROOTMacros.cmake in GENERATE_LIBRARY, seems to add the ${DEPENDENCIES} (implicitly) PUBLIC, which will also add them to any user of that library. I do not think this is a good default behavior, but whatever. 

One way to solve this would be to add a PRIVATE in front of every dependency, another to fix ROOTMacros.cmake in FairRoot so all the stuff gets added private. 

The cmake-y thing to do, I feel, was to use ${Boost_REGEX_LIBRARY} (note the unique capitalization), which resolves to the full path of the relevant lib (plus possibly some [additional keywords](https://cmake.org/cmake/help/v3.15/module/FindBoost.html). 

Here is to hoping that it will compile on MacOS as well.